mcs/class/System.Windows.Forms/System.Windows.Forms/TabControl.cs

```diff
@@ -1715,31 +1715,32 @@ public override void Add (Control value)
 
 			public override void Remove (Control value)
 			{
-				bool change_index = false;
+				bool invalid_selected_index = false;
 				
 				TabPage page = value as TabPage;
+				int next_selected_index = -1;
 				if (page != null && owner.Controls.Contains (page)) {
 					int index = owner.IndexForTabPage (page);
-					if (index < owner.SelectedIndex || owner.SelectedIndex == Count - 1)
-						change_index = true;
+
+					if (owner.SelectedIndex < index || owner.SelectedIndex == 0)
+						next_selected_index = owner.SelectedIndex;
+					else
+						next_selected_index = --owner.SelectedIndex;
```

Comment:
This has a side effect of calling the `SelectedIndex` setter, use `next_selected_index = owner.SelectedIndex - 1;` instead.

Reply:
Done!